File: java/src/main/java/it/giannibombelli/wsc2026/payment/infrastructure/providers/PayPal.java
package it.giannibombelli.wsc2026.payment.infrastructure.providers;

import it.giannibombelli.wsc2026.common.domain.primitive.Money;
import it.giannibombelli.wsc2026.common.domain.primitive.Timestamp;
import it.giannibombelli.wsc2026.payment.domain.ports.PaymentProvider;
import it.giannibombelli.wsc2026.payment.domain.ports.PaymentProviderResult;

import java.util.UUID;

public final class PayPal implements PaymentProvider {
    @Override
    public PaymentProviderResult process(UUID paymentId, UUID providerReference, Money amount) {
        return new PaymentProviderResult.Success(UUID.randomUUID(), Timestamp.now());
    }

    @Override
    public PaymentProviderResult refund(UUID paymentId, UUID providerReference, Money amount) {
        return new PaymentProviderResult.Success(UUID.randomUUID(), Timestamp.now());
    }
}

File: typescript/src/payment/infrastructure/providers/payPal.ts
// Adapter per il provider PayPal (stub di successo).

import { Timestamp } from '@/common/domain/primitive/timestamp.js';
import { Uuid } from '@/common/domain/primitive/uuid.js';
import type { Money } from '@/common/domain/primitive/money.js';
import type { PaymentProvider } from '../../domain/ports/paymentProvider.js';
import { PaymentProviderResult, paymentProviderSuccess } from '../../domain/ports/paymentProviderResult.js';

export class PayPal implements PaymentProvider {
  process(_paymentId: Uuid, _providerReference: Uuid, _amount: Money): PaymentProviderResult {
    return paymentProviderSuccess(Uuid.generate(), Timestamp.now());
  }

  refund(_paymentId: Uuid, _providerReference: Uuid, _amount: Money): PaymentProviderResult {
    return paymentProviderSuccess(Uuid.generate(), Timestamp.now());
  }
}
